### PR TITLE
Feature/sep6 32 create endpoint for movies where the actors played include the played movies in actor

### PR DESCRIPTION
**Pull request by tomondre**
Goal of the PR was to fetch only specific information when fetching all movies, so that the loading time does not take too long. There was a workaround created for the naming convention if jpa method

### CODE REVIEW

**Comment by tomondre** [comment]
I am looking at the refactor to create optional fields

**Comment by tomondre** [comment]
@KhaledHammoun Please check if there are any problems with the merge, cuz I had to manually merge a ton of code

**Comment by tomondre** [comment]
> I am looking at the refactor to create optional fields

Done

File: backend/src/main/java/com/sep6/backend/repository/ActorsRepository.java [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
 
 public interface ActorsRepository {
-    List<Person> getActors();
-    List<Person> getActorsBySearch(String search);
-    Optional<Person> getActorById(int id);
+    List<PersonProjection> getActors();
+    List<PersonProjection> getActorsBySearch(String search);
     Person save(Person person);
-
+    PersonMoviesProjection getActorById(int id);
```
Comment (KhaledHammoun) [inline backend/src/main/java/com/sep6/backend/repository/ActorsRepository.java:14]: It is missing the Optional here

File: backend/src/main/java/com/sep6/backend/repository/ActorsRepositoryImpl.java [hunk 2]
```diff
@@ -16,30 +18,29 @@ public class ActorsRepositoryImpl implements ActorsRepository
...
     @Override
-    public List<Person> getActorsBySearch(String search) {
+    public List<PersonProjection> getActorsBySearch(String search) {
         return jpaRepository.findAllByTypeAndNameContainingIgnoreCase(PersonType.ACTOR, search);
     }
 
     @Override
-    public Optional<Person> getActorById(int id) {
-        return jpaRepository.findByTypeAndId(PersonType.ACTOR, id);
+    public PersonMoviesProjection getActorById(int id) {
```
Comment (KhaledHammoun) [inline backend/src/main/java/com/sep6/backend/repository/ActorsRepositoryImpl.java:32]: Optional missing

**Review by KhaledHammoun — CHANGES_REQUESTED** [review]

**Review by KhaledHammoun — APPROVED** [review]

**Pull request merged**